staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/clientset/versioned/clientset.go

```diff
@@ -63,6 +63,9 @@ func NewForConfig(c *rest.Config) (*Clientset, error) {
 		}
 		configShallowCopy.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(configShallowCopy.QPS, configShallowCopy.Burst)
 	}
+	if c.Timeout == 0 {
+		klog.Warning("The provided config doesn't specify a timeout. The Timeout specifies a time limit for requests made by this client. Request without timeout can hang forever. Usually, it is a bad idea.")
```

Comment:
apiserver has a default timeout of 60s covering all non-long-running requests, i think a suggested client-side timeout should be greater other it will interfere w/ the server-side timeout.

Reply:
Not sure if we should suggest a concrete value. It's up to the caller. Some clients might require tight values, for example, https://github.com/kubernetes/kubernetes/pull/96217